SourceX/main.cpp

```diff
@@ -22,8 +22,20 @@ static std::string build_cmdline(int argc, char **argv)
 	return str;
 }
 
+void quit(){
+	static bool completed = false;
+	if(completed) return;
+	completed = true;
+	dvl::init_cleanup();
+	dvl::SaveGamma();
+	SDL_Quit();
+}
+
 int main(int argc, char **argv)
 {
+	atexit(quit);
 	auto cmdline = build_cmdline(argc, argv);
-	return dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);
+	int result = dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);
+	quit();

```

Comment:
won't this make it call quit twice since it is also called at exit?

Reply:
You say some platforms like android not support atexit and it not called.

Reply:
atexit() is called, but it is seen as a bad practice to rely on exit().